### ISSUE TITLE
"Slug already exists" warning does not appear if slug has only numbers in it, while creating new taxonomy.

### ISSUE THREAD

**Issue by sathyapriya91** [TaxoPress/TaxoPress#744]
1. Select Taxopress -> Taxonomies
2. Select Add New-> Taxonomy slug-> Enter a slug name that has only numbers that already exists (123)
3. Enter plural, singular label
4. Select any post types and select Add Taxonomy

"Please choose a different taxonomy name. 123(slug) is already registered." message appears.

It would be nice if the warning message "Slug already exists" appears immediately after entering the taxonomy slug. Instead of allowing the user to enter all the mandatory fields and the user gets to know that they can't create the taxonomy only after entering all the fields.

![123 slug](https://user-images.githubusercontent.com/83974581/128416264-8467e1cc-7d48-4bb0-8332-581631e79f64.JPG)
![123](https://user-images.githubusercontent.com/83974581/128416269-26e32f7e-5938-404d-a708-66107c489a89.JPG)
![taxonomy](https://user-images.githubusercontent.com/83974581/128416270-80a55f4d-4170-4287-be0d-e1c62ffbcab7.JPG)

**Comment by olatechpro** [comment]
Hi @stevejburge , this issue ended up been a big deal than i thought as i spend the good part of my Saturday debugging.

Will try to explain the issue the best way you get get it, when taxonomy is registered using the function `register_taxonomy` and the slug is integer, WordPress safe the slug as index for example, '0' => '65345643', '1' => '47646443' depending on it position instead of 'media_tag' => 'media_tag', 'post_tags' => 'post_tags' and primarily, most people fetching for existing taxonomy will get the keys and that's the reason the instant slug exist is not showing.

Of course, i initially fixed this by getting each key value instead but i realise there's another problem for when these taxonomy is added to a post. It's generating `PHP Notice:  Trying to get property 'term_id'` and some other notice both in admin post list and when user visit the post page and these error are coming from WordPress core because of the same integer issue.

In summary, the problem is the way WordPress store taxonomy that the slug is integer either registered through TaxoPress or code and as a result, the solution i end up implementing is to prevent using number only as taxonomy slug by showing a warning to user when they try to create Taxonomy.
